### PR TITLE
Persistence layer (#268)

### PR DESCRIPTION
**Pull request by vlad-tokarev**
for https://github.com/odahu/odahu-flow/issues/268

### CODE REVIEW

File: packages/operator/controllers/stub_clients.go [hunk 1]
```diff
@@ -0,0 +1 @@
+package controllers
```
Comment (keshamin) [inline packages/operator/controllers/stub_clients.go:1]: let's remove it :)

File: packages/operator/pkg/repository/packaging/repository.go [hunk 2]
```diff
@@ -14,39 +14,56 @@
...
 	"github.com/odahu/odahu-flow/packages/operator/pkg/utils"
 )
 
 const (
 	TagKey = "name"
 )
 
+type DeprecatedRepository interface {
+	Service
+	ResultRepository
```
Comment (keshamin) [inline packages/operator/pkg/repository/packaging/repository.go:32]: `Service` already includes `ResultRepository`. I think we can remove this line.

File: packages/operator/pkg/appserver/workers/v1/packaging/model_packaging.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+func NewSyncer(storage packaging.Repository, service packaging.Service) *Syncer {
+	return &Syncer{
+		storage: storage,
+		service: service,
+	}
+}
+
+func (mts *Syncer) AttachController(mgr ctrl.Manager, rec reconcile.Reconciler) error {
```
Comment (keshamin) [inline packages/operator/pkg/appserver/workers/v1/packaging/model_packaging.go:83]: Minor one - I guess mts is for Model Training Syncer. So mps should be for Model Packaging Syncer. The same is applicable to Model Deployment Syncer.
```suggestion
func (mps *Syncer) AttachController(mgr ctrl.Manager, rec reconcile.Reconciler) error {
```

File: packages/operator/pkg/testhelpers/stubclients/stub_clients.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+func (t TIStubClient) GetToolchainIntegration(name string) (*training.ToolchainIntegration, error) {
+	entity, ok := t.db[name]
+	if !ok {
+		return nil, errors.NotFoundError{Entity: name}
+	}
+	return &training.ToolchainIntegration{
+		ID:     name,
+		Spec:   entity.Spec,
+		Status: entity.Status,
+	}, nil
```
Comment (keshamin) [inline packages/operator/pkg/testhelpers/stubclients/stub_clients.go:29]: As I understand it is the same to the following. Isn't it?
```suggestion
	return &entity, nil
```

File: packages/operator/pkg/testhelpers/stubclients/stub_clients.go [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+func (t TIStubClient) UpdateToolchainIntegration(md *training.ToolchainIntegration) error {
+	panic("implement me")
+}
+
+func (t TIStubClient) CreateToolchainIntegration(md *training.ToolchainIntegration) error {
+	t.db[md.ID] = training.ToolchainIntegration{
+		ID:    md.ID,
+		Spec:  md.Spec,
+		Status: md.Status,
+	}
```
Comment (keshamin) [inline packages/operator/pkg/testhelpers/stubclients/stub_clients.go:54]: Is it necessary to re-create the `ToolchainIntegration` structure?
```suggestion
	t.db[md.ID] = *md
```

File: packages/operator/controllers/modeltraining_controller.go [hunk 6]
```diff
@@ -59,51 +57,31 @@ type ModelTrainingReconciler struct {
...
-	toolchainRepo   train_repository.ToolchainRepository
+	resultStorage   train_repository.ResultRepository
+	tiHTTPClient    train_repository.ToolchainRepository
 	trainingConfig  config.ModelTrainingConfig
 	operatorConfig  config.OperatorConfig
 	gpuResourceName string
 }
 
-// newReconciler returns a new reconcile.Reconciler
+// newReconciler returns a new reconcile.Reco nciler
```
Comment (keshamin) [inline packages/operator/controllers/modeltraining_controller.go:67]: misprint fix
```suggestion
// newReconciler returns a new reconcile.Reconciler
```

**Review by keshamin — APPROVED** [review]

**Review by maxkochubey — APPROVED** [review]

**Pull request merged**